sharding/shard.go

```diff
@@ -0,0 +1,219 @@
+package sharding
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+type shardBackend interface {
+	Get(k common.Hash) ([]byte, error)
```

Comment:
Can you comment on these exported methods? 
Why does this return an error but none of the other methods can return an error?